pkg/ingester/ingester.go

```diff
@@ -743,10 +735,11 @@ func (i *Ingester) PushWithCleanup(ctx context.Context, pushReq *push.Request) (
 		return nil, wrapWithUser(err, userID)
 	}
 
-	if err := db.acquireAppendLock(); err != nil {
+	lockState, err := db.acquireAppendLock(req.MinTimestamp())
```

Comment:
optimization tip: distributor already had to iterate through samples in the request, and could perhaps include minimum timestamp in the request passed to ingester.

Reply:
We could, but I doubt we would see any real difference. I don't expect `MinTimestamp()` to show up in profiles at all.